weather_sp/splitter_pipeline/file_splitters.py

```diff
@@ -93,29 +93,31 @@ def split_data(self) -> None:
             return
 
         with self._open_grib_locally() as grbs:
-            for grb in grbs:
-                splits = dict()
-                for dim in self.output_info.split_dims():
-                    try:
-                        splits[dim] = getattr(grb, dim)
-                    except RuntimeError:
-                        self.logger.error(
-                            'Variable not found in grib: %s', dim)
-                key = self.output_info.formatted_output_path(splits)
-                if key not in outputs:
-                    outputs[key] = FileSystems.create(key)
-                outputs[key].write(grb.tostring())
-                outputs[key].flush()
-
-            for out in outputs.values():
-                out.close()
+            try:
```

Comment:
why is this needed? is  there an indication that is raises an exception here and the output files are left dangling? Or is this just in case? I'd rather not add just-in-case try-blocks if they may not ever be needed.

Reply:
This offers a guarantee that the `close()` operations are called during any possible error. I don't have evidence that this is a problem now, but it's prudent to include. If I didn't do this, I would refactor this section of code to use a `with` expression (when I tried this, I found that it caused changes of data). 

I wouldn't say that this is never needed: the `finally` block is called every time this method is run -- including when there are exceptions.

Reply:
Using with here seems preferable, but I guess this works too.

Reply:
I just stepped into this code through the debugger. I now have a better understanding of what's going on. This pattern now makes sense to me: What I was missing before is that the `write` grib message step happens multiple times per output file. So, we would either have to a) keep a in-memory buffer of all the grib messages that will eventually get flused, then use a `with` expression at the end to write it once, or b) do what we're doing now. 

I'll leave a comment here that tries to explain this, for future reference.